server/matchmaker/matchmaker_queue.py

```diff
@@ -177,23 +184,30 @@ async def find_matches(self) -> None:
             except Exception:
                 self._logger.exception("Match callback raised an exception!")
 
-    def find_teams(self) -> List[Search]:
-        searches = []
-        unmatched = list(self._queue.keys())
-        need_team = []
-        for search in unmatched:
-            if len(search.players) == self.team_size:
-                searches.append(search)
-            else:
-                need_team.append(search)
-
-        if all(len(s.players) == 1 for s in need_team):
-            teams, unmatched = make_teams_from_single(need_team, self.team_size)
-        else:
-            teams, unmatched = make_teams(need_team, self.team_size)
-        searches.extend(teams)
-
-        return searches
+    def _register_unmatched_searches(
+        self,
+        searches: List[Search],
+        matches: List[Match]
+    ):
+        """
+        Tells all unmatched searches that they went through a failed matching
+        attempt.
+        """
+        searches_set = set(searches)
+        matched_searches = set(
+            search
+            for match in matches
+            for combinedSearch in match
+            for search in combinedSearch.get_original_searches()
+        )
+        unmatched_searches = searches_set - matched_searches
```

Comment:
Maybe it would be better to define the matchmaker interface to return both a list of matches and a list of unmatched searches, like how it was with the `make_teams` functions?

Reply:
Good point